sqlite.go

```diff
@@ -391,8 +396,19 @@ func (s *stmt) NumInput() int {
 }
 
 func (s *stmt) Close() error {
-	if s.persist {
-		return s.reserr("Stmt.Close", s.resetAndClear())
+	// We return this statement to the conn only if it's persistent, and
+	// only if there's not already a statement with the same query already
+	// cached there.
+	shouldPersist := s.persist
+	if _, alreadyPersisted := s.conn.stmts[s.query]; alreadyPersisted {
```

Comment:
no need to check this map if shouldPersist is true.

that penalizes conns that have a mix of persist and non-persist queries (barely)